parquet-variant/src/main/java/org/apache/parquet/variant/VariantUtil.java

```diff
@@ -329,6 +366,75 @@ static Variant.Type getType(ByteBuffer value) {
     }
   }
 
+  /**
+   * Computes the actual size (in bytes) of the Variant value at `value[pos...]`
+   * @param value The Variant value
+   * @param pos The starting index of the Variant value
+   * @return The size (in bytes) of the Variant value
+   */
+  public static int valueSize(ByteBuffer value, int pos) {
```

Comment:
I think this can be avoided by a different serialization strategy for arrays and objects because you could use the actual bytes written for the length of elements or field values.